dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -91,85 +91,207 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
       }
+    } yield oracleEventOpt
+  }
+
+  def getEvent(descriptor: EventDescriptorTLV): Future[Option[OracleEvent]] = {
+    eventDAO.findByDescriptor(descriptor).map { dbs =>
+      if (dbs.isEmpty) {
+        None
+      } else Some(OracleEvent(dbs))
     }
   }
 
-  def createNewEvent(
+  def createNewLargeRangedEvent(
+      eventName: String,
+      maturationTime: Instant,
+      base: UInt16,
+      isSigned: Boolean,
+      numDigits: Int): Future[OracleAnnouncementTLV] = {
+    require(base > UInt16.zero,
+            s"base cannot be less than 1, got ${base.toInt}")
+    require(numDigits > 0, s"numDigits cannot be less than 1, got $numDigits")
+
+    val descriptorTLV = if (isSigned) {
+      UninitializedLargeRangeEventDescriptorV0(base, isSigned, numDigits + 1)
+    } else UninitializedLargeRangeEventDescriptorV0(base, isSigned, numDigits)
+
+    createNewEvent(eventName, maturationTime, descriptorTLV)
+  }
+
+  def createNewRangedEvent(
       eventName: String,
       maturationTime: Instant,
-      outcomes: Vector[String]): Future[EventDb] = {
+      start: Int,
+      count: Int,
+      step: Int): Future[OracleAnnouncementTLV] =
+    createNewRangedEvent(eventName,
+                         maturationTime,
+                         Int32(start),
+                         UInt32(count),
+                         UInt16(step))
+
+  def createNewRangedEvent(
+      eventName: String,
+      maturationTime: Instant,
+      start: Int32,
+      count: UInt32,
+      step: UInt16): Future[OracleAnnouncementTLV] = {
+    require(count > UInt32.zero,
+            s"Count cannot be less than 1, got ${count.toInt}")
+    require(step > UInt16.zero,
+            s"Step cannot be less than 1, got ${step.toInt}")
+
+    val descriptorTLV =
+      UninitializedRangeEventDescriptorV0(start, count, step)
+
+    createNewEvent(eventName, maturationTime, descriptorTLV)
+  }
+
+  def createNewEnumEvent(
+      eventName: String,
+      maturationTime: Instant,
+      outcomes: Vector[String]): Future[OracleAnnouncementTLV] = {
     require(outcomes.nonEmpty, "Cannot make an event with no outcomes")
     require(outcomes.distinct.size == outcomes.size,
             s"Cannot have duplicate outcomes, got $outcomes")
+
+    val descriptorTLV = UninitializedEnumEventDescriptorV0(outcomes)
+
+    createNewEvent(eventName, maturationTime, descriptorTLV)
+  }
+
+  def createNewEvent(
+      eventName: String,
+      maturationTime: Instant,
+      uninitializedDescriptor: UninitializedEventDescriptor[
+        EventDescriptorTLV]): Future[OracleAnnouncementTLV] = {
+    require(maturationTime.isAfter(TimeUtil.now),
+            s"Event cannot mature in the past, got $maturationTime")
+
     for {
       indexOpt <- rValueDAO.maxKeyIndex
-      index = indexOpt match {
+      firstIndex = indexOpt match {
         case Some(value) => value + 1
         case None        => 0
       }
 
       signingVersion = SigningVersion.latest
```

Comment:
Maybe make `signingVersion` a parameter? You can provide a default value if you want (`SigningVersion.latest`). I just think this makes testing much easier.